### PR TITLE
Update references to New Relic's language-specific exporters for OpenTelemetry

### PR DESCRIPTION
**Pull request by jgals**
See DOC-6641 for details.

This PR updates references to New Relic's language-specific exporters for OpenTelemetry in favor of the collector and OTLP.

### CODE REVIEW

**Comment by urbiz-nr** [comment]
Thanks for the updates, @jgals - could you sign the CLA so I can approve the changes?

**Comment by jgals** [comment]
Thank you, @urbiz-nr! I have signed the CLA and will wait on a content review from you and @a-feld at this point.

File: src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-architecture-recipes.mdx
Comment (a-feld) [inline src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-architecture-recipes.mdx:51]: I think this needs to be a separate option. There are 2 options here getting shmoosed into 1 😄 

Options:
1. Use infinite tracing tail-based sampling
2. Use OpenTelemetry-Collector based tail-based sampling

Comment (jgals) [reply src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-architecture-recipes.mdx]: Thanks for the feedback, @a-feld. I've separated these options for tail-based sampling.

**Comment by urbiz-nr** [comment]
Hi @jgals, @a-feld - is this ready for review? Thanks!

**Review by a-feld — APPROVED** [review]

**Comment by jgals** [comment]
Hi @urbiz-nr, yes, this is ready for review. Thank you!

**Review by urbiz-nr — APPROVED** [review]

**Review by urbiz-nr — APPROVED** [review]

**Review by urbiz-nr — APPROVED** [review]

File: src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx
Comment (RichVanderwal) [inline src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:162]: The OpenTelemetry project's OTLP exporters for each language are in various states of development, and at least Java and Python's are GA. Perhaps you mean that it's our New Relic *OTLP endpoint* that's in pre-release?

File: src/content/docs/integrations/open-source-telemetry-integrations/troubleshooting/troubleshoot-opentelemetry-exporter.mdx
Comment (RichVanderwal) [inline src/content/docs/integrations/open-source-telemetry-integrations/troubleshooting/troubleshoot-opentelemetry-exporter.mdx:20]: The link for the "New Relic exporter for Go" points to the Exporter that's being deprecated. Link to the [New Relic Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/newrelicexporter) in the collector instead -- these are two different pieces of software.

In fact, two paragraphs above, the phrase "If you're using a New Relic OpenTelemetry exporter" makes it sound like this is a troubleshooting guide for the exporters we're deprecating. Does it make sense to have this page at all?
Pasted_Image_4_22_21__8_49_AM

File: src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx
Comment (jgals) [reply src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:162]: Yes, good point @rvanderwal-newrelic, I'llfollow-up with a fix PR to replace _OTLP exporter_ with _OTLP endpoint_.

Comment (jgals) [reply src/content/docs/integrations/open-source-telemetry-integrations/opentelemetry/introduction-opentelemetry-new-relic.mdx:162]: https://github.com/newrelic/docs-website/pull/1975

File: src/content/docs/integrations/open-source-telemetry-integrations/troubleshooting/troubleshoot-opentelemetry-exporter.mdx
Comment (jgals) [reply src/content/docs/integrations/open-source-telemetry-integrations/troubleshooting/troubleshoot-opentelemetry-exporter.mdx:20]: Thanks for the clarification, @rvanderwal-newrelic. I will update these links in a follow up PR.

Comment (jgals) [reply src/content/docs/integrations/open-source-telemetry-integrations/troubleshooting/troubleshoot-opentelemetry-exporter.mdx:20]: #1975

**Pull request merged**
